news/_posts/2022-01-05-multistore-configuration-forms.md

```diff
@@ -0,0 +1,69 @@
+---
+layout: post
+title: "Multistore configuration forms"
+subtitle: "How multistore configuration forms got better with PrestaShop 1.7.8"
+date: 2022-01-05 
+authors: [matthieu-rolland]
+icon: icon-puzzle
+tags: [multistore, contribute]
+---
+
+Did you know that PrestaShop has a cool feature called "multistore", which is the ability to manage multiple shops from the same back office ?
+
+One of the back-office aspects of this feature is that many configuration forms can have different values for different shops, which is dealt with using shop contexts.
+
+## How it works and what was less than stellar about it
+
+In a configuration form, a merchant can set different values for different shop contexts, of which there are three, in this order of inheritence:
+
+- All shop context
+- Group shop context
+- SIngle shop context
+
+![Shop contexts](/assets/images/2022/01/shopcontexts.png)
+
+When a value is not set for a context, it inherits the value of the parent context. Consequently, the "All Shops" context values can be considered a default values. 
+
+Unfortunately, before Prestashop 1.7.8, this mechanism was not obvious on the forms in question, and it was not easy to see at a glance which form fields were modified or not for the current context.
+
+Something had to be done in order to make this feature more understandable.
+
+## Objectives
+
+Our multistore implementation had to meet 2 main criterias:
+
+- Improve existing UX for better understanding
+
+- Make the technical implementation as easily reusable as possible on existing and future configuration forms, so external contributors and module developers can easily reuse our implementation in configuration forms.
+
+## How we did it
+
+### The UX
+
+After numerous merchant interviews, our UX team created mock-ups, then our product team wrote functional specifications in accordance with merchant feedback. 
```

Comment:
```suggestion
After numerous merchant interviews, UX designers team created mock-ups, then Product owners wrote functional specifications in accordance with merchant feedback. 
```
If we have public links we can share it would be great 👍